src/BizHawk.Emulation.Cores/Consoles/Nintendo/QuickNES/QuickNES.cs

```diff
@@ -82,21 +82,88 @@ public QuickNES(byte[] file, QuickNESSettings settings, QuickNESSyncSettings syn
 		private void SetControllerDefinition()
 		{
 			ControllerDefinition def = new("NES Controller");
+
+			// Function to add gamepad buttons
 			void AddButtons(IEnumerable<(string PrefixedName, uint Bitmask)> entries)
 				=> def.BoolButtons.AddRange(entries.Select(static p => p.PrefixedName));
-			AddButtons(_syncSettings.Port1 switch
+
+			// Parsing Port1 niputs
+
+			switch (_syncSettings.Port1)
 			{
-				Port1PeripheralOption.Gamepad => GamepadButtons[0],
-				Port1PeripheralOption.FourScore => FourScoreButtons[0],
-				_ => Enumerable.Empty<(string PrefixedName, uint Bitmask)>()
-			});
-			AddButtons(_syncSettings.Port2 switch
+				case Port1PeripheralOption.Gamepad:
+
+					// Adding set of gamepad buttons (P1)
+					AddButtons(GamepadButtons[0]);
+					
+					break;
+
+				case Port1PeripheralOption.FourScore:
+
+					// Adding set of gamepad buttons (P1)
+					AddButtons(FourScoreButtons[0]);
+
+					break;
+
+				case Port1PeripheralOption.ArkanoidNES:
+
+					// Adding Arkanoid Paddle potentiometer
+					def.AddAxis("P2 Paddle", 0.RangeTo(160), 80);
+
+					// Adding Arkanoid Fire button
+					def.BoolButtons.Add("P2 Fire");
+
+					break;
+
+				case Port1PeripheralOption.ArkanoidFamicom:
+
+					// Adding set of gamepad buttons (P1)
+					AddButtons(GamepadButtons[0]);
+
+					// Adding dummy set of P2 buttons (not yet supported)
+					def.BoolButtons.Add("P2 Up");
+					def.BoolButtons.Add("P2 Down");
+					def.BoolButtons.Add("P2 Left");
+					def.BoolButtons.Add("P2 Right");
+					def.BoolButtons.Add("P2 B");
+					def.BoolButtons.Add("P2 A");
+					def.BoolButtons.Add("P2 M"); // Microphone
+
+					// Adding Arkanoid Paddle potentiometer
+					def.AddAxis("P3 Paddle", 0.RangeTo(160), 80);
+
+					// Adding Arkanoid Fire button
+					def.BoolButtons.Add("P3 Fire");
+
+					break;
+
+				default: break;
+			}
+
+			// Parsing Port2 niputs
+
+			switch (_syncSettings.Port2)
 			{
-				Port2PeripheralOption.Gamepad => GamepadButtons[1],
-				Port2PeripheralOption.FourScore2 => FourScoreButtons[1],
-				_ => Enumerable.Empty<(string PrefixedName, uint Bitmask)>()
-			});
+				case Port2PeripheralOption.Gamepad:
+
+					// Adding set of gamepad buttons (P1)
+					AddButtons(GamepadButtons[1]);
+
+					break;
+
+				case Port2PeripheralOption.FourScore2:
+
+					// Adding set of gamepad buttons (P2)
+					AddButtons(FourScoreButtons[1]);
+
+					break;
+
+				default: break;

```

Comment:
this and the above default branch are not necessary.

Reply:
done